### ISSUE TITLE
Colons and rest of URL disappearing

### ISSUE THREAD

**Issue by matthewhartstonge** [a85/POSTMan-Chrome-Extension#3994]
## App Details:
```
Postman for Windows
Version 5.5.0
win32 10.0.16299 / x64
```

### 1. New bug or old?

I've just upgraded to 5.5.0 on Windows 10 x64.

For our RESTful APIs we follow Google API design spec, therefore operations that don't fit in the normal HTTP verbs use `/{resourceID}/:action`. Postman has started stripping the colon and following words  from the URL.

The same behaviour is described in issue #2805 which was closed in September.

### 2. Expected Behaviour
```
http://localhost:8080/v1/cats/c6621640-cde3-40c2-9e7e-db5dd89ec6f6/:feed
```

'Code' shows:
```
POST /v4/cats/c6621640-cde3-40c2-9e7e-db5dd89ec6f6/:feed HTTP/1.1
Host: localhost:8080
...
```

### 3. Console Logs

Trying to access `http://localhost:8090/v4/assessmenttemplates/c6621640-cde3-40c2-9e7e-db5dd89ec6f6/:apply` returns the following in the postman console

![capture](https://user-images.githubusercontent.com/6119549/34189546-d083b7a2-e5a0-11e7-9238-f80c25b9c047.PNG)

### 4. Steps to reproduce
Trying to access the following fails:
```
http://localhost:8080/v1/cats/c6621640-cde3-40c2-9e7e-db5dd89ec6f6/:feed
```

'Code' shows:
```
POST /v4/cats/c6621640-cde3-40c2-9e7e-db5dd89ec6f6/ HTTP/1.1
Host: localhost:8080
...
```

This shows up correctly via 'code' but fails:
```
http://localhost:8080/v1/cats/c6621640-cde3-40c2-9e7e-db5dd89ec6f6/::feed
```

'Code' shows:
```
POST /v4/cats/c6621640-cde3-40c2-9e7e-db5dd89ec6f6/:feed HTTP/1.1
Host: localhost:8080
```

Finally, URL Encoding the colon works:
```
http://localhost:8080/v1/cats/c6621640-cde3-40c2-9e7e-db5dd89ec6f6/%3Afeed
```

'Code' shows:
```
POST /v4/cats/c6621640-cde3-40c2-9e7e-db5dd89ec6f6/%3Afeed HTTP/1.1
Host: localhost:8080
```

**Comment by sivcan** [comment]
@MatthewHartstonge 
`:feed` in the url will be served as a **path variable** in the postman app and when you're trying to make the request, postman checks for it's value and since the values are undefined, the app truncates it. 

To get this working there are two ways : 
1. Add a variable to your collection with the key as "feed" and the value as ":feed" 
Your url will look like so : 
`http://localhost:8080/v1/cats/c6621640-cde3-40c2-9e7e-db5dd89ec6f6/{{feed}}`
Then on making the request the correct URL will be hit. 

2. Add params to your request by clicking on params.
Add the key as "feed" and the value as ":feed" and now on hitting the normal URL, your request will go through.

Docs : https://www.getpostman.com/docs/postman/environments_and_globals/variables
Hope this solves the issue.

**Comment by matthewhartstonge** [comment]
Hi there,

Thanks for that. I had a quick look at the docs provided and there was no mention of Path Variables. I did find them mentioned in [Requests](https://www.getpostman.com/docs/postman/sending_api_requests/requests) though. It may be worth updating the variable docs to include this for other people who abide by the [Google RESTful API Design spec](https://cloud.google.com/apis/design/).

It was also pointed out to me that Postman could make the default param value useful. For example, one easy way to work around this issue, so as not to confuse others, is to default the param to itself with a colon. Such as the following:

![capture](https://user-images.githubusercontent.com/6119549/34231154-ba0ddc38-e63f-11e7-926c-e43d7223f874.PNG)

Since postman can use both `:var` and `{{var}}` yet `{{var}}` works in all cases, it may make more sense to simply deprecate `:var` in favour of `{{var}}` to avoid confusion and breaking of APIs.

Another issue I just thought of is that `:var` will break is any API that may deal with and work with URNs - that is authorisation endpoints.

For a quick and dirty example: 
```
POST /v1/authorization/arn:aws:iam::123456789012:user/David
{
  "id": "arn:aws:iam::123456789012:user/David",
  "username": "david@example.com"
...
}
```

**Comment by matthewhartstonge** [comment]
Hi @sivcan, what's the road map on this as I wouldn't consider it fixed.

**Comment by prashantagarwal** [comment]
@MatthewHartstonge We are planning to deprecate path variables in future and hence it is recommended to use Postman variables wherever possible.

**Comment by matthewhartstonge** [comment]
Sweet, thanks! That's awesome to know 😄

**Comment by bmoeskau** [comment]
Doesn't seem like this answered the fundamental question about properly supporting the ":" character within the path of a URL. We also use the Google syntax guidelines, specifically the [Custom Methods](https://cloud.google.com/apis/design/custom_methods) recommendation of using a colon to separate custom `name:verb` resource names (e.g. separating `POST /foo` (create) vs `POST /foo:search` (to support passing complex filters for the same resource type via POST body).

This is the worst kind of issue, in that the behavior is both unexpected, and fails silently when triggered. So moving toward variables is fine and good, but can you also fix the **bug** in how paths are parsed?

**Comment by vchernogorov** [comment]
So this scenario still doesn't work: `https://photoslibrary.googleapis.com/v1/albums/:album-id:batchAddMediaItems`. It converts `:album-id:batchAddMediaItems` to the path parameter instead of `:album-id` only. Although `https://photoslibrary.googleapis.com/v1/albums/{{album-id}}:batchAddMediaItems` is working as expected.

**Comment by syats** [comment]
For anyone coming here for an actual solution or workaround to this bug, you have to replace the colons in the URL's for   %3A    (percent, three, capitaA) characters.

**Comment by bmoeskau** [comment]
Just another comment that one of our customers hit this again today, and it's super annoying since the outcome is a random 404. So I guess we're left with:
- Documenting clearly that if you use Postman, you have to know to replace `:` with `%3A` (and handling the inevitable issues)
- Telling our customers NOT to use Postman to avoid confusion (which of course is not reasonable)
- Changing the (valid) semantics of our API to avoid potential issues like this (again, *just* for Postman)

Or of course, option D) where y'all just handle this character correctly, which is what I would vote for. It would be one thing if this was open source, but we are actually paying money for this.

**Comment by codenirvana** [comment]
Hey Everyone, apologize that this bug got unnoticed for so long.
We've pushed a fix for this in the latest Postman app `v8.10.0`.

![Screenshot 2021-08-05 at 6 30 44 PM](https://user-images.githubusercontent.com/4865836/128356317-6a56040e-38e3-4737-8db9-dca0249345c6.png)

Please update your app and confirm if this fixes your issue?
